src/libraries/System.IO.FileSystem.AccessControl/tests/MemoryMappedFileAclTests.cs

```diff
@@ -0,0 +1,532 @@
+        //[InlineData(MemoryMappedFileRights.AccessSystemSecurity)]
+        //public void CreateFromFile_VerifySecurity(MemoryMappedFileRights rights)
+        //{
+        //    using TempFile file = new TempFile(GetTestFilePath(), StandardCapacity);
+        //    MemoryMappedFileSecurity security = GetSecurity(rights, AccessControlType.Allow);
+        //    using MemoryMappedFile mmf = CreateFromFile(file, security: security);
+        //}
+
+
+        // CreateNew
+
+        [Fact]
+        public void CreateNew_MapNameEmpty()
+        {
+            Assert.Throws<ArgumentException>(() => CreateNew(mapName: string.Empty));
+        }
+
+        [Theory]
+        [InlineData(-1)]
+        [InlineData(0)]
+        public void CreateNew_CapacityOutOfRange(long capacity)
+        {
+            AssertExtensions.Throws<ArgumentOutOfRangeException>("capacity", () => CreateNew(capacity: capacity));
+        }
+
+        [ConditionalFact(typeof(MemoryMappedFileAclTests), nameof(MemoryMappedFileAclTests.IsAddressSpaceWithIntPtrSize4))]
+        public void CreateNew_CapacityLargerThanLogicalAddressSpace()
+        {
+            AssertExtensions.Throws<ArgumentOutOfRangeException>("capacity", () => CreateNew(capacity: long.MaxValue));
+        }
+
+        [Theory]
+        [InlineData((MemoryMappedFileAccess)(-1))]
+        [InlineData((MemoryMappedFileAccess)(6))]
+        public void CreateNew_AccessOutOfRange(MemoryMappedFileAccess access)
+        {
+            AssertExtensions.Throws<ArgumentOutOfRangeException>("access", () => CreateNew(access: access));
+        }
+
+        [Fact]
+        public void CreateNew_AccessWriteNotAllowed()
+        {
+            AssertExtensions.Throws<ArgumentException>("access", () => CreateNew(access: MemoryMappedFileAccess.Write));
+        }
+
+        [Theory]
+        [InlineData((MemoryMappedFileOptions)(-1))]
+        [InlineData((MemoryMappedFileOptions)(1))]
+        [InlineData((MemoryMappedFileOptions)(67108865))]
+        public void CreateNew_OptionsOutOfRange(MemoryMappedFileOptions options)
+        {
+            AssertExtensions.Throws<ArgumentOutOfRangeException>("options", () => CreateNew(options: options));
+        }
+
+        [Theory]
+        [InlineData((HandleInheritability)(-1))]
+        [InlineData((HandleInheritability)(2))]
+        public void CreateNew_InheritabilityOutOfRange(HandleInheritability inheritability)
+        {
+            AssertExtensions.Throws<ArgumentOutOfRangeException>("inheritability", () => CreateNew(inheritability: inheritability));
+        }
+
+        [Fact]
+        public void CreateNew_SecurityNull()
+        {
+            AssertExtensions.Throws<ArgumentNullException>("memoryMappedFileSecurity", () => CreateNew(security: null));
+        }
+
+        [Fact]
+        public void CreateNew_MapNameNull()
+        {
+            using MemoryMappedFile mmf = CreateNew(mapName: null);
+            Assert.NotNull(mmf);
+        }
+
+        //[Theory]
```

Comment:
I'm investigating why this unit test is not passing. The security object extracted via `GetAccessControl` is not the same as the one I passed to the `Create*` methods.